src/ufo.rs

```diff
@@ -114,22 +147,32 @@ impl Default for MetaInfo {
 }
 
 impl Ufo {
-    /// Crate a new `Ufo`.
+    /// Create a new `Ufo`.
     pub fn new() -> Self {
         Ufo::default()
     }
 
+    /// Create a new `Ufo` only with certain fields
+    pub fn with_fields(data_request: UfoDataRequest) -> Self {
```

Comment:
I might just have a `set_data_request` method? Alternatively there could be a builder-style method, like:

```rust
impl Ufo {
    // fn new etc
    pub fn with_data_request(mut self, data_request: UfoDataRequest) -> Self {
        self.data_request = data_request;
        self
    }
}

// and we use this like,

let my_ufo = Ufo::new().with_data_request(UfoDataRequest::default());
```